google/auth/_cloud_sdk.py

```diff
@@ -83,14 +80,49 @@ def get_application_default_credentials_path():
     return os.path.join(config_path, _CREDENTIALS_FILENAME)
 
 
+def _get_active_config(config_path):
+    """Gets the active config for the Cloud SDK.
+
+    Args:
+        config_path (str): The Cloud SDK's config path.
+
+    Returns:
+        str: The active configuration name.
+    """
+    active_config_filename = os.path.join(config_path, 'active_config')
+
+    if not os.path.isfile(active_config_filename):
+        return 'default'
+
+    with io.open(active_config_filename, 'r', encoding='utf-8') as file_obj:
+        active_config_name = file_obj.read().strip()
```

Comment:
No porcelain, and the CLI changes it's face too often to be dependable. However, since it has to maintain backwards-compatibility with older versions, the config files are relatively stable. Finally, if any of this code fails the worst-case scenario is that we can't detect the project ID.